tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -223,9 +232,32 @@ private static ArgumentParser argParser() {
                 .dest("producerConfigFile")
                 .help("producer config properties file.");
 
+        parser.addArgument("--print.metrics")
+                .action(storeTrue())
+                .type(Boolean.class)
+                .metavar("PRINT-METRICS")
+                .dest("printMetrics")
+                .help("print out metrics at the end of the test.");
+
         return parser;
     }
 
+    private static void printMetrics(KafkaProducer producer) {
+        Map<MetricName, ? extends Metric> metrics = producer.metrics();
+        int maxLengthOfDisplayName = 0;
+        for (Metric metric : metrics.values()) {
+            MetricName mName = metric.metricName();
+            String mergedName = mName.group() + ":" + mName.name();
```

Comment:
The tags in the MetricName can be useful too. For example, for the per node metrics, it tells us the broker id.